### ISSUE TITLE
Question about GenD checkpoint conversion and unusually high AUC on CDF-v2

### ISSUE THREAD

**Issue by dinhxuanhuy** [yermandy/GenD#5]
Dear Author,

First of all, thank you for releasing your implementation. I am currently working on a paper and have been experimenting with your GenD model.

I converted your released .ckpt weight file into a .pth format so that I could use it within a DeepFakeBench-formatted reimplementation of GenD. After running evaluation on CDF-v2 with 32-frame video testing, I obtained a video-level AUC of 96.7%, which seems higher than expected.

I would like to ask whether this result is reasonable, or if I may have made a mistake during the weight conversion or evaluation process.

For reference:

My GenD reimplementation (DeepFakeBench format):
https://github.com/thanhquan123hi1/DeepfakeBench/blob/main/training/detectors/gend_detector.py
The checkpoint conversion code I used is shown below:
Could you please let me know:

Whether this checkpoint mapping logic looks correct?

Whether a 96.7% video-level AUC on CDF-v2 (32 frames) is expected rather than the code that use average from 5 seed on your paper? 

def convert_gend_ckpt_to_detector(ckpt_path, save_path):

    ckpt = torch.load(ckpt_path, map_location="cpu")

    if "state_dict" in ckpt:
        old_state_dict = ckpt["state_dict"]
    else:
        old_state_dict = ckpt

    new_state_dict = {}

    for key, value in old_state_dict.items():
        new_key = key

        if "vision_model" in key:
            parts = key.split("vision_model.")
            if len(parts) > 1:
                layer_name = parts[1]
                new_key = f"backbone.{layer_name}"
                new_key = new_key.replace(".base_layer", "")
                new_key = new_key.replace(".ln_tuning_layers.default", "")

        elif "model.linear" in key:
            new_key = key.replace("model.linear", "head")

        if new_key != key:
            print(f"Mapping: {key} -> {new_key}")

        new_state_dict[new_key] = value

    torch.save(new_state_dict, save_path)

ckpt_path = "/content/DeepfakeBench/model.ckpt"
pth_path = "/content/DeepfakeBench/model_gend.pth"
convert_gend_ckpt_to_detector(ckpt_path, pth_path)

**Comment by yermandy** [comment]
@dinhxuanhuy 96.7% is a possible value, it can happen that these 32 samples were a bit "easier" than my 32 samples. At the same time, there can be a variation in detection.

**Comment by yermandy** [comment]
@dinhxuanhuy You can try evaluating on more datasets and see the discrepancy.
